entrypoints/services/AppStateService.ts

```
@@ -0,0 +1,65 @@
+            } else {
+                // Update existing document
+                await updateDoc(stateRef, dataToSave);
+            }
+
+            return newState;
+        } catch (error) {
+            console.error("Error saving app state:", error);
+            throw error;
+        }
+    }
+}
```

Biome (1.9.4)

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)